src/middlewared/middlewared/schema.py

```diff
@@ -585,10 +587,17 @@ def clean(self, value):
         value = super(List, self).clean(value)
         if value is None:
             return copy.deepcopy(self.default)
-        if not isinstance(value, list):
+        if not isinstance(value, (list, tuple)) and not issubclass(type(value), list):
```

Comment:
What is the `value` for which `isinstance(value, (list, tuple))` is `False`, but `issubclass(type(value), list)` is `True`?

Reply:
query-filter type. Come to think of it, might be something wonky going on with our schema stuff...

Reply:
Sorry, what's query-filter type? What python type is it?

Reply:
Hmm... can't seem to reproduce the issue I was seeing that caused this. As I was looking at the general problem again, I realized the PR could be simplified and improved by just re-using validation used in filter_list.